### DISCUSSION TITLE
Docker job container always shows "tail -f /dev/null" as entrypoint, even though container's Dockerfile specifies otherwise

### DISCUSSION THREAD

**Discussion by mmic-bjohnson** [community/community#41644] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello!

I have a GitHub Action that specifies a custom Docker images in the `jobs.[myjob].container.image` key.

The `Dockerfile` for said image contains

```
COPY entrypoint.sh /scripts/entrypoint.sh
RUN ["chmod", "+x", "/scripts/entrypoint.sh"]
ENTRYPOINT ["/bin/sh", "-c", "/scripts/entrypoint.sh"]
```

The referenced `entrypoint.sh` script contains

```
#!/bin/bash

/usr/bin/chromedriver &
/usr/local/sbin/php-fpm
```

If I start this container in any other Docker environment, the `docker ps` output shows the entrypoint to be `"/scripts/entrypoint.sh"`, but when the same image is used for the `container` in this Action, the entrypoint is always shown to be `"tail -f /dev/null"`, which seems like some kind of fallback to prevent the container from exiting under any circumstance.

Is this expected behavior, i.e., is the Docker image's entrypoint overridden when the image is used for the `container` in this context? I don't seem to have this problem for other custom Docker images that are used as `services` in the same `job`, as those always show the correct/expected `entrypoint` values.

The documentation at  https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions does not seem to mention this behavior.

Thanks in advance for any guidance here!

**Comment by airtower-luna** [comment, marked as answer]
> Is this expected behavior, i.e., is the Docker image's entrypoint overridden when the image is used for the container in this context?

It is, you can see the Docker command used to start the container in the job log. The job container is used only as a build environment, without starting anything else. The documentation for [`jobs.<job_id>.container`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontainer) doesn't explicitly mention it, that could certainly be improved.

> I don't seem to have this problem for other custom Docker images that are used as services in the same job, as those always show the correct/expected entrypoint values.

That's exactly the difference: The service containers are meant to run services for the build.
